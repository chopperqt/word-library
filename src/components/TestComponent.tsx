import { useState } from "react";

const TestComponent = () => {
  const [value, setValue] = useState('')

  return (
    <div data-testid="test-field-wrap">
      <input
        value={value}
        onChange={(e) => setValue(e.target.value)}
        data-testid='test-field'
      />
      {value === '123' && (
        <div data-testid='test-field-error'>
          Test
        </div>
      )}
    </div>
  );
}

export default TestComponent;