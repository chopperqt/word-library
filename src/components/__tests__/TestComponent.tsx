import {
  cleanup,
  fireEvent,
  render,
  screen,
} from '@testing-library/react'

import '@testing-library/jest-dom/extend-expect'
import TestComponent from 'components/TestComponent'

afterEach(cleanup)

it('Test Change', async () => {
  render(<TestComponent />)

  const field = screen.getByTestId('test-field')

  expect(field).toBeTruthy()

  fireEvent.change(field, {
    target: {
      value: '123',
    }
  })

  expect(field).toHaveValue('123')

  const error = await screen.findByTestId('test-field-error')

  expect(screen.getByTestId('test-field-wrap')).toMatchSnapshot()
})