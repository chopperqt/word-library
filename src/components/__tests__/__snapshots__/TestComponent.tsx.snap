// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Change 1`] = `
<div
  data-testid="test-field-wrap"
>
  <input
    data-testid="test-field"
    value="123"
  />
  <div
    data-testid="test-field-error"
  >
    Test
  </div>
</div>
`;
